---
title: "atac_tdg_test"
author: "VF"
date: "22 4 2022"
output: html_document
---
### Pre-processing of Atac-Tdg-Samples with PL script run on Kg_tdg_ko_1

#### Copy of files to raw folder
```{r}
getwd()
setwd("/mnt/bohacek/kathi/vincent/")
file.copy(from = "X204SC21114463.Z01.F003/X204SC21114463-Z01-F003/raw_data/Kg_tdg_ko_2/Kg_tdg_ko_2_EKDL220003833-1a_HJ5T2DSX3_L1_2.fq.gz",to = "ATAC_tdg_Ko/raw/")
file.copy(from = "X204SC21114463.Z01.F003/X204SC21114463-Z01-F003/raw_data/Kg_tdg_ko_3/Kg_tdg_ko_3_EKDL220003834-1a_HJ5T2DSX3_L1_2.fq.gz",to = "ATAC_tdg_Ko/raw/")
file.copy(from = "X204SC21114463.Z01.F003/X204SC21114463-Z01-F003/raw_data/Kg_tdg_M_1/Kg_tdg_M_1_EKDL220003835-1a_HJ5T2DSX3_L1_2.fq.gz",to = "ATAC_tdg_Ko/raw/")
file.copy(from = "X204SC21114463.Z01.F003/X204SC21114463-Z01-F003/raw_data/Kg_tdg_M_3/Kg_tdg_M_3_EKDL220003836-1a_HJ5T2DSX3_L1_2.fq.gz",to = "ATAC_tdg_Ko/raw/")
file.copy(from = "/mnt/bohacek/kathi/vincent/test_ATAC/raw/Kg_tdg_ko_1_EKDL220000948-1a_H22J7DSX3_L2_1.fq.gz",to = )
file.copy(from = "/mnt/bohacek/kathi/vincent/test_ATAC/raw/Kg_tdg_ko_1_EKDL220000948-1a_H22J7DSX3_L2_2.fq.gz",to = "/mnt/bohacek/kathi/vincent/ATAC_tdg_Ko/raw/")
```




### script for pre-processing wrote by PL 

```{bash, eval = False}
#!/bin/bash

ref=/reference/Mus_musculus/Ensembl/GRCm38/Sequence/BOWTIE2Index/genome
adapters=/conda/share/trimmomatic-0.38-1/adapters/TruSeqPE.fa
mkdir -p trimmed
mkdir -p aligned
mkdir -p tracks

for f in raw/*_1.fq.gz; do

base=`basename $f _1.fq.gz`
bam=aligned/"$base".bam

echo $base
trimdir=trimmed

if [ -f "$trimdir/"$base"_1.paired.fastq.gz" ]; then
  echo $trimdir/$base"_*fq.gz found; skipping"
else
trimmomatic PE -threads 6 -summary $trimdir/"$base".stats -phred33 raw/"$base"_1.fq.gz raw/"$base"_2.fq.gz $trimdir/"$base"_1.paired.fastq.gz $trimdir/"$base"_1.unpaired.fastq.gz $trimdir/"$base"_2.paired.fastq.gz $trimdir/"$base"_2.unpaired.fastq.gz ILLUMINACLIP:$adapters:2:15:4:4:true LEADING:20 TRAILING:20 SLIDINGWINDOW:4:15 MINLEN:25
fi

if [ -f "$bam" ]; then
    echo "$bam found; skipping"
else
(bowtie2 -p 6 --dovetail --no-mixed --no-discordant -I 15 -X 2000 -x $ref -1 $trimdir/"$base"_1.paired.fastq.gz -2 $trimdir/"$base"_2.paired.fastq.gz) 2> aligned/"$base".bowtie2 | samtools view -bS - | samtools sort -@4 -m 2G - > $bam

java -jar /common/picard.jar MarkDuplicates I=$bam O=$base.bam.2 M=aligned/$base.picard.dupMetrics.txt && mv $base.bam.2 $bam

samtools index $bam && rm $trimdir/"$base"_*.fastq.gz
fi

if [ -f "tracks/$base.bw" ]; then
    echo "tracks/$base.bw found; skipping"
else
  bamCoverage -p 6 --ignoreDuplicates --effectiveGenomeSize 2652783500 --normalizeUsing CPM --ignoreForNormalization MT -b aligned/"$base".bam -o tracks/$base.bw
fi

done
```

```{bash, generating tracks for NFbam not done}

mkdir -p NFtracks
for f in NF/*.shifted.bam; do
base=`basename $f .shifted.bam`
bam=NFbam/"$base".shifted.bam

echo $base
if [ -f "NFtracks/$base.bw" ]; then
    echo "NFtracks/$base.bw found; skipping"
else
bamCoverage -p 6 --ignoreDuplicates --effectiveGenomeSize 2652783500 --normalizeUsing CPM --ignoreForNormalization MT -b NFbam/"$base".shifted.bam -o NFtracks/$base.bw
fi
done
```


### script for peak calling ATAC nucleosome free and nucleosome occupied

```{bash, eval =False}
#!/bin/bash

blacklist=/reference/Mus_musculus/mm10.blacklist.bed
mkdir -p NFbam
mkdir -p NFpeaks
mkdir -p peaks

for f in aligned/*.bam; do
base=`basename $f .bam`
echo $base
bam2=NFbam/$base.shifted.bam
if [ -f "$bam2" ]; then
  echo "$bam2 found, skipping..."
else
  alignmentSieve --blackListFileName $blacklist --ATACshift --filterMetrics NFbam/$base.filterMetrics --minMappingQuality 20 \
    --ignoreDuplicates --minFragmentLength 40 --maxFragmentLength 115 -p 8 -b $f -o /dev/stdout | samtools sort -@ 4 -m 1G - > $bam2
  samtools index $bam2
fi
if [ -f NFpeaks/$base"_peaks.narrowPeak" ]; then
 echo "Peaks found for $base, skipping..."
else
 macs2 callpeak --call-summits -g mm -f BAMPE --name NFpeaks/$base -t $bam2
  macs2 callpeak --broad -g mm -f BAMPE --name peaks/$base -t $f
fi
done



```
### script for calling total accessibility 
```{bash}
#!/bin/bash

blacklist=/reference/Mus_musculus/mm10.blacklist.bed
mkdir -p total_bam
mkdir -p Broadpeaks


for f in aligned/*.bam; do
base=`basename $f .bam`
echo $base
#bam2=total_bam/$base.shifted.bam
#if [ -f "$bam2" ]; then
 # echo "$bam2 found, skipping..."
#else
 # alignmentSieve --blackListFileName $blacklist --ATACshift --filterMetrics total_bam/$base.filterMetrics --minMappingQuality 20 \
  #  --ignoreDuplicates -p 8 -b $f -o /dev/stdout | samtools sort -@ 4 -m 1G - > $bam2
  #samtools index $bam2
#fi
if [ -f Broadpeaks/$base"_peaks.broadPeak" ]; then
 echo "Peaks found for $base, skipping..."
else
 macs2 callpeak --broad  -g mm -f BAMPE --name Broadpeaks/$base -t $f
  
fi
done

```

```{bash engine.opts='-l'}
fastqc -o aligned --threads 4 aligned/Kg_tdg_ko_2_EKDL220003833-1a_HJ5T2DSX3_L1.bam        
```
### script for pre-processing wrote by PL for merged samples (not done yet)

```{bash, eval = False}
#!/bin/bash

ref=/reference/Mus_musculus/Ensembl/GRCm38/Sequence/BOWTIE2Index/genome
adapters=/conda/share/trimmomatic-0.38-1/adapters/TruSeqPE.fa
mkdir -p m_trimmed
mkdir -p m_aligned
mkdir -p m_tracks

for f in raw/*_1.fq.gz; do

base=`basename $f _1.fq.gz`
bam=m_aligned/"$base".bam

echo $base
trimdir=m_trimmed

if [ -f "$trimdir/"$base"_1.paired.fastq.gz" ]; then
  echo $trimdir/$base"_*fq.gz found; skipping"
else
trimmomatic PE -threads 6 -summary $trimdir/"$base".stats -phred33 raw/"$base"_1.fq.gz raw/"$base"_2.fq.gz $trimdir/"$base"_1.paired.fastq.gz $trimdir/"$base"_1.unpaired.fastq.gz $trimdir/"$base"_2.paired.fastq.gz $trimdir/"$base"_2.unpaired.fastq.gz ILLUMINACLIP:$adapters:2:15:4:4:true LEADING:20 TRAILING:20 SLIDINGWINDOW:4:15 MINLEN:25
fi

if [ -f "$bam" ]; then
    echo "$bam found; skipping"
else
(bowtie2 -p 6 --dovetail --no-mixed --no-discordant -I 15 -X 2000 -x $ref -1 $trimdir/"$base"_1.paired.fastq.gz -2 $trimdir/"$base"_2.paired.fastq.gz) 2> m_aligned/"$base".bowtie2 | samtools view -bS - | samtools sort -@4 -m 2G - > $bam

java -jar /common/picard.jar MarkDuplicates I=$bam O=$base.bam.2 M=m_aligned/$base.picard.dupMetrics.txt && mv $base.bam.2 $bam

samtools index $bam && rm $trimdir/"$base"_*.fastq.gz
fi

if [ -f "m_tracks/$base.bw" ]; then
    echo "m_tracks/$base.bw found; skipping"
else
  bamCoverage -p 6 --ignoreDuplicates --effectiveGenomeSize 2652783500 --normalizeUsing CPM --ignoreForNormalization MT -b vm_aligned/"$base".bam -o m_tracks/$base.bw
fi

done
```



